cmd/ipfs/init.go

```diff
@@ -237,9 +237,11 @@ func initializeIpnsKeyspace(repoRoot string) error {
 	}
 	defer nd.Close()
 
-	err = nd.SetupOfflineRouting()
-	if err != nil {
-		return err
+	if nd.PrivateKey == nil {
+		err = nd.LoadPrivateKey()
```

Comment:
I'd say that this is cheap enough to not have to do this lazily too

Reply:
I'm more worried about not *having* the key. We have quite a few tests ATM where we don't specify one.

(on the other hand, I'm fine making it a requirement)

Reply:
Doing a bit of local end-to-end testing, that .1s is lost in the noise.